drivers/gpio/gpio_aw9523b.c

```diff
@@ -0,0 +1,504 @@
+	if (err) {
+		LOG_ERR("%s: Failed to configure interrupt %d (%d)", dev->name,
+			config->int_gpio.pin, err);
+		return err;
+	}
+
+	gpio_init_callback(&data->gpio_callback, config->int_cb, BIT(config->int_gpio.pin));
+	err = gpio_add_callback(config->int_gpio.port, &data->gpio_callback);
+	if (err) {
+		LOG_ERR("%s: Failed to add interrupt callback for pin %d (%d)", dev->name,
+			config->int_gpio.pin, err);
+		return err;
+	}
+
+end_init_int_gpio:
+#endif
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(reset_gpios)
+	if (!config->reset_gpio.port) {
+		goto end_hw_reset;
+	}
+
+	if (!gpio_is_ready_dt(&config->reset_gpio)) {
+		LOG_ERR("%s: Reset GPIO not ready", dev->name);
+		return -ENODEV;
+	}
+
+	err = gpio_pin_configure_dt(&config->reset_gpio, GPIO_OUTPUT_ACTIVE);
+	if (err) {
+		LOG_ERR("%s: Failed to configure reset pin %d (%d)", dev->name,
+			config->reset_gpio.pin, err);
+		return err;
+	}
+
+	k_busy_wait(AW9523B_RESET_PULSE_WIDTH);
+
+	err = gpio_pin_set_dt(&config->reset_gpio, 0);
+	if (err) {
+		LOG_ERR("%s: Failed to set 0 reset pin %d (%d)", dev->name, config->reset_gpio.pin,
+			err);
+		return err;
+	}
+
+end_hw_reset:
+#endif
+
+	if (!device_is_ready(config->i2c.bus)) {
+		return -ENODEV;
+	}
+
+	k_sem_init(aw9523b_get_lock(config->mfd_dev), 1, 1);
+
+	/* Software reset */
+	err = i2c_reg_read_byte_dt(&config->i2c, AW9523B_REG_SW_RSTN, buf);
+	if (err) {
+		LOG_ERR("%s: Failed to software reset (%d)", dev->name, err);
+		return err;
+	}
+
+	/* Disabling all interrupts */
+	err = i2c_burst_write_dt(&config->i2c, AW9523B_REG_INT0, int_init_data, sizeof(buf));
+	if (err) {
+		LOG_ERR("%s: Failed to disable all interrupts (%d)", dev->name, err);
+		return err;
+	}
+
+	if (!config->port0_push_pull) {
+		/* Configure port0 to push-pull mode */
+		err = i2c_reg_update_byte_dt(&config->i2c, AW9523B_REG_CTL, AW9523B_GPOMD, 0xFF);
+		if (err) {
+			LOG_ERR("%s: Failed to configure port0 to push-pull (%d)", dev->name, err);
+			return err;
+		}
+	}
+
+	return 0;
+}
+
+#define GPIO_AW9523B_DEFINE(inst)                                                                  \
+	static struct gpio_aw9523b_data gpio_aw9523b_data##inst;                                   \
+                                                                                                   \
+	IF_ENABLED(DT_INST_PROP_HAS_IDX(inst, int_gpios, 0), (                                     \
+		static void gpio_aw9523b_int_cb##inst(const struct device *dev,                    \
+							     struct gpio_callback *cb,             \
+							     gpio_port_pins_t pins)                \
+		{                                                                                  \
+			k_work_submit(&gpio_aw9523b_data##inst.intr_worker);                       \
+		}                                                                                  \
+	))                                                                                         \
```

Comment:
I don't think you need to generate this, can you do

```c
static __maybe_unused void gpio_aw9523b_int_handler(const struct device *gpio_dev,
                                   struct gpio_callback *cb,
                                   uint32_t pins)
{       
        struct gpio_aw9523b_data *data = CONTAINER_OF(
                        cb, struct gpio_aw9523b_data, gpio_callback); 
                
        k_work_submit(&data->intr_worker);
}
```
instead? (copied it from input_paw32xx.c)

Reply:
I worried about whether __maybe_unused can be used. It is standardized in C23, not C99.
If I can use it, it is better to use it.

Reply:
It's pretty widely used in the code base, I wouldn't worry about it.

Reply:
I got it. I applied your suggestion.

Reply:
ok I see you added it on other callbacks but would you not use it to replace `gpio_aw9523b_int_cb##inst` with a common one as well?

Reply:
Ah, I misunderstood what you said.
This is better. I've addressed it.